Modules/_sqlite/connection.c

```diff
@@ -370,6 +370,13 @@ pysqlite_connection_close_impl(pysqlite_Connection *self)
         return NULL;
     }
 
+    if (!self->initialized) {
+        pysqlite_state *state = pysqlite_get_state(NULL);
+        PyErr_SetString(state->ProgrammingError,
+                        "Base Connection.__init__ not called.");
+        return NULL;
+    }
```

Comment:
We cannot use `pysqlite_check_connection` here, since it also checks `self->db`. Calling `close()` twice should still be a no-op.